### ISSUE TITLE
[Badge]: accessibility issues related to the component

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#1463]
### Description
Role of "In Progress" dropdown is incorrectly determined.

### Precondition
Screen Reader is **ON**

### Steps to Reproduce
1. Open [Badge](https://uui.epam.com/documents?id=badge&mode=doc&skin=UUI4_promo&category=components) component.
2. Use the `Tab` key to navigate to the "In Progress" dropdown and note announcement.

### Actual Result
"In Progress, button" is announced.

### Expected Result
Screen reader should announce role "combobox"  (e.g. In Progress, combobox). 
**Reference for implementation:** https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/
**Failed WCAG Checkpoint:** 4.1.2 Name, Role, Value (Level A) 
https://www.w3.org/TR/WCAG21/#name-role-value
